src/warc2zim/url_rewriting.py

```diff
@@ -152,46 +289,69 @@ def get_without_fragment(url: str) -> str:
 class ArticleUrlRewriter:
     """Rewrite urls in article."""
 
-    def __init__(self, article_url: str, known_urls: set[str]):
+    def __init__(self, article_url: HttpUrl, existing_zim_paths: set[ZimPath]):
+        self.article_path = normalize(article_url)
         self.article_url = article_url
-        self.known_urls = known_urls
-        self.base_path = f"/{urlsplit(normalize(article_url)).path}"
-        if self.base_path[-1] != "/":
-            # We want a directory
-            self.base_path = posixpath.dirname(self.base_path)
```

Comment:
I'm surprised you removed it.
It was necessary (to me at least) as posixpath and url differ a bit in they relative path computation.
(And it is probably the root (technical) cause of the other change in tests)

Reply:
From my experience, this was again needed only for root URL which were missing the trailing `/` in ZIM path (e.g. `http://kiwix.org` which were wrongly stored at `kiwix.org` instead of `kiwix.org/`)